posthog/queries/insight.py

```diff
@@ -1,17 +1,31 @@
-from typing import Optional
+from typing import Any, Dict, Optional
 
 from posthog.clickhouse.query_tagging import tag_queries
 from posthog.client import query_with_columns, sync_execute
+from posthog.models.instance_setting import get_instance_setting
+from posthog.settings.utils import get_list
 from posthog.types import FilterType
 
 
 # Wrapper around sync_execute, adding query tags for insights performance
 def insight_sync_execute(query, args=None, *, query_type: str, filter: Optional["FilterType"] = None, **kwargs):
-    _tag_query(query, query_type, filter)
+
+    kwargs["settings"] = add_team_specific_settings(settings=kwargs.get("settings", {}), filter=filter)
+
+    _tag_query(query, query_type, filter, settings=kwargs["settings"])
 
     return sync_execute(query, args=args, **kwargs)
 
 
+def add_team_specific_settings(settings: Dict[str, Any], filter: Optional["FilterType"] = None) -> Dict[str, Any]:
```

Comment:
Can you add a test for `PARALLEL_HASH_ENABLED_TEAMS` with `@snapshot_clickhouse_queries` on to make sure this works properly (and keeps working)?